### DISCUSSION TITLE
Ordered list of links

### DISCUSSION THREAD

**Discussion by thipokch** [isar/isar#715] [Help]
How would you go about ordered list of IsarLink? 

Let's imagine, we have a list of `Recipe`. Each recipe has ordered list of cooking`Step`. Each step has a video content showing you the cooking techniques. 

- You wouldn't use embedded objects, since you want to **reuse content for each step**. 
- You wouldn't use sort, since **each recipe has different order of operations**.

I wonder how would you go about this? Another example could be a music playlist.

Anyway, thanks for this awesome library!

**Comment by wildsurfer** [comment]
Same question 🤔

**Comment by ganeshh123** [comment]
Has anybody managed to find any solutions for this?

**Comment by basselalsayed** [comment]
assuming order is a property on your step model, I think this should work? This should avoid the need to call `recipe.steps.load()`. You need to call `filter` to gain access to queries

For my use case I had to go with a manual sort as I was doing something funky, but this should be fine if you're just reading. 

```
await recipe.steps.filter().sortByOrder().findAll();

return recipe;
```
